src/guides/build-iapp/advanced/result-callback.md

```diff
@@ -0,0 +1,182 @@
+---
+title: Result Callback Guide
+description:
+  Use the iExec result callback to have the protocol invoke a function on your
+  smart contract at the end of a (optionally confidential TEE) task execution.
+---
+
+# Result Callback
+
+This guide explains how to trigger a callback function at the end of a
+successful task on your smart contract.
+
+Use a callback when your smart contract should:
+
+- Ingest off-chain computed data (API aggregation, ML inference, analytics) and
+  persist it
+- React to an execution outcome (conditional trigger, state transition)
+- Store a timestamped record (price feed, score, KPI, proof hash)
+- Bridge logic between external systems and on-chain state
+
+## 🧩 High-Level Flow
+
+1. A requester executes an iApp.
+2. The iApp writes `${IEXEC_OUT}/computed.json` with a `callback-data` field
+   (ABI‑encoded bytes you crafted).
+3. After the task completes and is validated, the iExec protocol invokes your
+   contract’s `receiveResult(bytes32,bytes)`.
+4. Your contract decodes and processes those bytes if callback data have been
+   provided.
+
+## Step-by-Step Implementation
+
+### Step 1: Prepare the Callback Payload in the iApp
```

Comment:
I would invert step 1 and step 2 because it makes more sense to deploy the callback contract first.